### PR TITLE
Retain all CloudFormation resources in Deletion of CloudFormationStack

### PR DESCRIPTION
**Pull request by stephanlindauer**
This would be my suggestion on how to deal with CloudFormationStacks that get stuck in _**Deletion failed**_ state.
By retaining all resources that belong to the CloudFormationStacks we delete **just** the CloudFormationStack resource and **don't** cause cascading deletions. 
IMHO this should be the desired behavior. As with other resources, we just try to delete the resource itsself **EXcluding** the onces that depend on it.
The resources that were created from the CloudFormationStack and are left over after the CloudFormationStack is gone, should be deleted by their own resource implementation.
What do you think? 

ref: https://github.com/rebuy-de/aws-nuke/issues/238
ping @rebuy-de/prp-aws-nuke

### CODE REVIEW

**Comment by bjoernhaeuser** [comment]
The job of aws-nuke is to delete resources. Retaining them to delete them makes no sense to me. CloudFormation is a special snowflakes on aws, because it takes care of deleting the created resource for you. Why not leverage this to make it easier?

My proposal is to delete stacks first, wait for them to be deleted, and afterwards delete the rest of the config.

**Comment by stephanlindauer** [comment]
@bjoernhaeuser 

> The job of aws-nuke is to delete resources

excactly. but it would do it in either case. in theory it shouldn't make a difference whether we utilize the cloudformation deletion feature or whether we delete them from their own resource implementation. the result is the same: stuff is deleted.
the "deleting resources outside of CFS"-approach would enable us on top to use the config to protect resources that would otherwise be deleted.

> CloudFormation is a special snowflakes on aws, because it takes care of deleting the created resource for you. Why not leverage this to make it easier?

because this has turned out to be tricky in practice:
- we would have to squeeze this feature into aws-nuke to have this one exception
- if a resource depends on something that was created via CFS then the deletion will fail (i can give you some examples from our own infrastructure where this would bite us, later in person)
- if the deletion fails the resource in CFS that caused it might get deleted by plain old aws-nuke resource deletion. so the next time we come around trying to delete the stack it will fail again because it tries to delete something that isn't there anymore. et voila: endless faillure loop. 

TL:DR: i do agree that deleting CFS without utilizing the deletion feature feels a bit odd, but weight against it's alternatives, i think it's what would make most sense for our purpose (and i think of the users of this tool as well).
